sources/shared-defaults/kubernetes-containerd-nvidia.toml

```diff
@@ -4,4 +4,30 @@ template-path = "/usr/share/templates/containerd-config-toml_k8s_nvidia_containe
 
 # Image registries
 [metadata.settings.container-registry]
-affected-services = ["containerd", "host-containers", "bootstrap-containers"]
+affected-services = ["containerd", "host-containers", "bootstrap-containers", "soci-snapshotter"]
+
+[configuration-files.snapshotter-toml]
+path = "/etc/containerd/config.d/001-snapshotter.toml"
+template-path = "/usr/share/templates/snapshotter-toml"
+overwrite-path-if-present = false
+
+# Container runtime - soci.
+[services.soci-snapshotter]
+configuration-files = ["soci-config-toml"]
+restart-commands = ["/bin/systemctl try-restart soci-snapshotter.service"]
+
+[configuration-files.soci-config-toml]
+path = "/etc/soci-snapshotter/config.toml"
+template-path = "/usr/share/templates/soci-config-toml"
+
+[configuration-files.selected-snapshotter]
+path = "/etc/containerd/selected-snapshotter"
+template-path = "/usr/share/templates/selected-snapshotter"
+
+# Container runtime settings.
+
+[metadata.settings.container-runtime]
+affected-services = ["containerd", "soci-snapshotter"]
```

Comment:
```suggestion
affected-services = ["containerd"]
```

Reply:
Actually, `[metadata.settings.container-runtime]` is already covered in [defaults.toml](https://github.com/bottlerocket-os/bottlerocket/blob/54e53fdc7299cb0379c0e711b4a80ab62b4f47da/sources/shared-defaults/defaults.toml#L32-L36) . This should be removed from both the shared `kubernetes-containerd{-nvidia}.toml` files - we don't want changes to `settings.container-runtime` to start soci-snapshotter if it wasn't configured, nor does it need to be restarted if it is configured when any of these settings change.